lib/screens/profile_onboarding.dart

```diff
@@ -162,26 +166,56 @@ class _ProfileOnboardingState extends State<ProfileOnboarding> {
     );
   }
 
-  void _validateAndSubmit() {
+  void _validateAndSubmit() async {
     if (!_formKey.currentState.validate()) {
       return;
     }
     _formKey.currentState.save();
 
-    _birthDate = DateTime.parse('$_birthYear-$_birthMonth-$_birthDay');
+    _birthDate = DateTime.parse(
+        '$_birthYear-$_birthMonth-$_birthDay'); // we're not even using this
+    User _firebaseUser = EmailAuth().getCurrentUser();
+    ProfileOnboardingService _profileOnboardingService =
+        new ProfileOnboardingService();
 
-    print('$_role, $_firstName, $_lastName, $_city, $_birthDate');
+    /// Create the user object to be sent out.
+    UserModel user = new UserModel(
+      id: _firebaseUser.uid,
+      firstName: _firstName,
+      lastName: _lastName,
+      firebaseUuid: _firebaseUser.uid,
+      email: _firebaseUser.email,
+      city: _city,
+      category: [], //we don't have form field for this
+      pictureUrl: '',
+      role: _role,
+      ratings: [],
+      averageRating: 0.0,
+    );
 
-    widget.isProfile
-        ? Navigator.pop(context)
-        : Navigator.push(
-            context,
-            MaterialPageRoute(
-              builder: (context) => SideMenu(
-                title: 'Project Eureka',
-              ),
-            ),
-          );
+    /// Using the profile onboarding serivce, send data to backend
+    try {
+      if (widget.isProfile) {
+        // Using HTTP PUT to update instead
+        // This currently doesn't work until we can get the current user_id
+        await _profileOnboardingService.updateUser(_firebaseUser.uid, user);
+
+        Navigator.pop(context);
+      } else {
+        // Using HTTP POST to add new user
+        await _profileOnboardingService.addUser(user);
+
+        Navigator.pushAndRemoveUntil(
+          context,
+          MaterialPageRoute(
+            builder: (context) => Home(),
+          ),
```

Comment:
When I click the `Confirm` button nothing happens but when I hot reload, it takes me to the home screen.

Reply:
I am having the same issue as well. When I fill out all the information and click Confirm I am not routed to home screen.

Reply:
I just pushed a new commit that hopefully will try to fix the problem both of you are having, can you guys confirm?